start/manage/update.php

```diff
@@ -17,57 +13,117 @@
 
 $ship = $shipLoader->findOneById($id);
 
+
 if ($_SERVER['REQUEST_METHOD'] == 'POST'){
-    header('location:index.php');die;
+
+
+    $ship->setName($_POST['ship']);
+    $ship->setWeaponPower($_POST['weaponPower']);
+    $ship->setJediFactor($_POST['jediFactor']);
+    $ship->setStrength($_POST['strength']);
+    $ship->setDescription($_POST['description']);
+
+
+    if (empty(trim($ship->getName()))){
+        $errmessages[] = "Please enter ship name";
+    }
+
+    if (empty(trim($ship->getWeaponPower()))){
+        $errmessages[] = "Please enter weapon power";
+    } elseif ($ship->getWeaponPower() < 0){
```

Comment:
I think these two elseif checks can be combined and just have a generic message like: 'Invalid number'. Also, take a look at the php method `is_numeric` instead of is_int.

This can be applied to the Jedi factor and strength checks too.

Reply:
Test